File: Java/1823_Find_the_Winner_of_the_Circular_Game.java
class Solution {
    public int findTheWinner(int n, int k) {
        List<Integer> list = new ArrayList<>();
        for(int i = 1; i <= n; i++){
            list.add(i);
        }
        //System.out.println(list);
        int temp = k;
        while(list.size() > 1){
            temp %= list.size();
            temp = temp == 0 ? list.size(): temp; 
            list.remove(temp-1);
            temp = temp-1 + k;
            //System.out.println(list);
        }
        return list.get(0);
    }
}

File: CS/1823_Find_the_Winner_of_the_Circular_Game.cs
public class Solution {
    public class ListNode{
        public int val;
        public ListNode next;
    }
    public int FindTheWinner(int n, int k) {
        ListNode head = new ListNode {val = 1};
        ListNode tail = head;
        ListNode nextNode;
        for(int i = 2; i <= n; i++){
            nextNode = new ListNode {val = i};
            tail.next = nextNode;
            tail = tail.next;
        }
        tail.next = head;
        
        while(head != tail){
            for(int i = 1; i < k; i++){
                head = head.next;
                tail = tail.next;
            }
            tail.next = head.next;
            head = tail.next;
        }
        return head.val;
    }
}
/*
1 2 3 4 6
        h
      t  
public class Solution {
    public int FindTheWinner(int n, int k) {
        List<int> list = new List<int>();
        int result = 0;
        for(int i = 1; i <= n; i++){
            list.Add(i);
        }
        int index = k-1;
        while(list.Count > 1){
            list.RemoveAt(index);
            index = (index + k-1) % list.Count;
        }
        return list[0];
    }
}
*/
